execution/group_by.go

```diff
@@ -151,18 +162,45 @@ func (gb *GroupByStream) AddRecord(ctx context.Context, tx storage.StateTransact
 		}
 	}
 
-	// Update aggregates
-	for i := range gb.aggregates {
-		var value octosql.Value
-		if gb.inputFields[i] == "*star*" {
-			mapping := make(map[string]octosql.Value, len(record.Fields()))
-			for _, field := range record.Fields() {
-				mapping[field.Name.String()] = record.Value(field.Name)
-			}
-			value = octosql.MakeObject(mapping)
-		} else {
-			value = record.Value(gb.inputFields[i])
+	// Create values and aggregates for the record
+	// This needs to be done here, because different records might result in different values
+	// from star expressions (i.e with different number of fields)
+
+	recordVariables := record.AsVariables()
+
+	fieldCount := len(gb.inputFields)
+
+	values := make([]octosql.Value, fieldCount)
+
+	for i, fieldName := range gb.inputFields {
+		value := record.Value(fieldName)
+		values[i] = value
+	}
+
+	outputFields := gb.originalOutputFields
+	for _, expr := range gb.starExpressions { // handle star expressions
+		starValues, starFields, err := expr.ValueAndName(ctx, recordVariables)
+		if err != nil {
+			return errors.Wrap(err, "couldn't get value of star expression")
 		}
+
+		outputFields = append(outputFields, starFields...)
+		values = append(values, starValues...)
+	}
+
+	gb.outputFields.extendByFields(outputFields)
+
+	// Add necessary aggregates
+	currentAggregateCount := len(gb.aggregates)
+	for i := 0; i < len(values)-currentAggregateCount; i++ {
```

Comment:
Might as well be `i := currentAggregateCount; i < len(values); i++`, could be a bit simpler